src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -475,6 +477,50 @@ def test_create_advanced_arb_sequence_wrong_size(self, session_5421):
         assert exc_info.value.args[0] == 'Length of marker_location_array and waveform_handles_array parameters do not match.'
         assert str(exc_info.value) == 'Length of marker_location_array and waveform_handles_array parameters do not match.'
 
+    # Multi-Threading tests
```

Comment:
All these tests are copied/pasted without any driver-specific customization.
Did you look into single sourcing?

Reply:
>All these tests are copied/pasted without any driver-specific customization.

Not all of them. `test_multi_threading_ivi_synchronized_wrapper_releases_lock` uses `commit` for some drivers and `abort` for those that don't have a `commit` function, though perhaps it could use `abort` for all of them.

>Did you look into single sourcing?

I didn't. Maybe I can shove them all into system_test_utilities.

Reply:
These tests are now single-sourced from system_test_utilities.